docs/architecture/blazor-for-web-forms-developers/security-authentication-authorization.md

```diff
@@ -1,15 +1,379 @@
+
+![apply migrations page](./media/security/apply-migrations-page.png)
+
+Click on the "Apply Migrations" button and the necessary database tables should be created for you. In addition, the migration files should appear in your project, as shown:
+
+![migration files](./media/security/migration-files.png)
+
+You can run the migration yourself, without running the web application, using this command line tool:
+
+```powershell
+dotnet ef database update
+```
+
+If you would rather run a script to apply the new schema to an existing database, you can script these migrations from the command line. Run this command to generate the script:
+
+```powershell
+dotnet ef migrations script -o auth.sql
+```
+
+This will produce a SQL script in the output file `auth.sql` which can then be run against whatever database you like. If you have any trouble running `dotnet ef` commands, [make sure you have the EF Core tools installed on your system](https://docs.microsoft.com/ef/core/miscellaneous/cli/dotnet).
+
+In the event you have additional columns on your source tables, you will need to identify the best location for these columns in the new schema. Generally, columns found on the `aspnet_Membership` table should be mapped to the `AspNetUsers` table. Columns on `aspnet_Roles` should be mapped to `AspNetRoles`. Any additional columns on the `aspnet_UsersInRoles` table would be added to the `AspNetUserRoles` table.
+
+It's also worth considering putting any additional columns on separate tables, so that future migrations won't need to take into account such customizations of the default identity schema.
+
+### Migrating data from universal providers to ASP.NET Core Identity
+
+Once you have the destination table schema in place, the next step is to migrate your user and role records to the new schema. A complete list of the schema differences, including which columns map to which new columns, can be found [here](https://docs.microsoft.com/aspnet/core/migration/proper-to-2x/membership-to-core-identity).
+
+To migrate your users from membership to the new identity tables, you should [follow the steps described in the documentation](https://docs.microsoft.com/aspnet/core/migration/proper-to-2x/membership-to-core-identity). After following these steps and the script provided, your users will need to change their password the next time they log in.
+
+It is possible to migrate user passwords but the process is much more involved. It's quite likely that many users' passwords from the prior system have previously been exposed along with their email address in hacks of other systems where the user used the same credentials. Requiring users to update their passwords as part of the migration process, and encouraging them to use new, unique passwords, is likely to enhance the overall security of the application.
```

Comment:
While I agree with you, I don't think we should presume hacked accounts on other systems :smile:

```suggestion
It is possible to migrate user passwords but the process is much more involved. Requiring users to update their passwords as part of the migration process, and encouraging them to use new, unique passwords, is likely to enhance the overall security of the application.
```